### DISCUSSION TITLE
Podman Quadlet stopping all of a sudden

### DISCUSSION THREAD

**Discussion by christian-2** [containers/podman#22313] [Q&A]
I have a container with a Java workload that I run as Podman Quadlet. When I add `java -Djavax.net.debug=ssl` (which will lead to substantially more output), the container stops all of a sudden and without any obvious reason (after writing approx. 50000 lines or 4.2M of output). There is sufficient space left in the file system. How can I find the root cause of what may be happening? I'm using Podman 4.7.2.

**UPDATE**  `journalctl` indicates that the Quadlet stops because of `NOPERMISSION`. If I run the Java workload without system property `javax.net.debug`, no stop occurs. So this is perhaps a combined Podman-Java "thing".

```
systemd[1]: my-quadlet.service: Main process exited, code=exited, status=4/NOPERMISSION
```

**Comment by eriksjolund** [comment]
I tried to reproduce the error you got.

```
systemd[1]: my-quadlet.service: Main process exited, code=exited, status=4/NOPERMISSION
```

### Experiment

1. Create a new user
   ```
   sudo useradd test
   ```
2. Open shell session for that user
   ```
   sudo machinectl shell --uid test
   ```
3. Create directory
   ```
   mkdir -p ~/.config/containers/systemd
   ```
4. Create the file _~/.config/containers/systemd/my-quadlet.container_ with the contents
   ```
   [Container]
   Image=docker.io/library/alpine
   Exec=sh -c "exit 4"
   ```
5. Reload the systemd configuration
   ```
   systemctl --user daemon-reload
   ```
6. Start the service
   ```
    systemctl --user start my-quadlet.service
   ```
7. Search for the error message
   ```
    journalctl -q --user -xeu my-quadlet.service | grep "status=4/NOPERMISSION"
   ```
   The command prints the following output
   ```
   Apr 10 07:34:40 localhost.localdomain systemd[24883]: my-quadlet.service: Main process exited, code=exited, status=4/NOPERMISSION
   ```
   __result:__ The output contains `my-quadlet.service: Main process exited, code=exited, status=4/NOPERMISSION`.
   In other words, the error message was reproduced.

### Discussion

As can be seen here

```
$ podman run --rm docker.io/library/alpine sh -c "exit 4"
$ echo $?
4
```

podman will exit with exit status 4 if the command exits with exit status 4.

The error message you got 
`my-quadlet.service: Main process exited, code=exited, status=4/NOPERMISSION`
indicates that the command in the container exited  with exit status 4.

>  So this is perhaps a combined Podman-Java "thing".

That is a possibility, but it could also be a pure Java "thing" as the experiment shows.


### About the system

Fedora CoreOS 40.20240331.1.0

```
$ podman --version
podman version 5.0.0
```

**Reply by christian-2** [reply]
Thanks you, this is very concrete and helpful.
